webdriver-spec.html

```diff
@@ -1435,288 +1435,532 @@
 
-<p>WebDriver <dfn>capabilities</dfn> allow the <a>local end</a>
- to specify what features it desires or requires the <a>remote end</a> to fulfill
- to be able to create a <a>new session</a>.
-
-<p>The following table provides an overview
- of recognised capabilities, and their usage and effect:
+<p>WebDriver <dfn>capabilities</dfn> are used to communicate the features
+supported by a given implementation. The <a>local end</a> may use capabilities
+to define which features it requires the <a>remote end</a> to satisfy when
+creating a <a>new session</a>. Likewise, the <a>remote end</a> uses capabilities
+to describe the full feature set for a session.
+
+<p>The following table provides an overview of the standard capabilities that
+each implementation <em>MUST</em> support. An implementation <em>MAY</em> define
+additional capabilities, although it is <em>RECOMMENDED</em> that custom
+capabilities prepend the capability key with a vendor prefix, as outlined in
+[[CSS21]].
+
+<aside class="example">
+<p>As an example, Mozilla could elect to hide new features behind capabilities
+with a "-moz-" prefix:
+
+<pre class=highlight>{
+  "browserName": "firefox",
+  "browserVersion": "1234",
+  "-moz-experimental-webdriver": true
+}</pre>
+</aside>
 
 <table class=simple>
- <tr>
+<thead>
+  <tr>
   <th>Capability
   <th>Key
   <th>Value Type
   <th>Description
- </tr>
+  </tr>
+</thead>
 
- <tr>
+<tbody>
+  <tr>
   <td><dfn>Browser name</dfn>
   <td>"<code>browserName</code>"
   <td>string
   <td>Identifies the user agent.
- </tr>
+  </tr>
 
- <tr>
+  <tr>
   <td><dfn>Browser version</dfn>
   <td>"<code>browserVersion</code>"
   <td>string
   <td>Identifies the version of the user agent.
- </tr>
+  </tr>
 
- <tr>
+  <tr>
   <td><dfn>Platform name</dfn>
   <td>"<code>platformName</code>"
   <td>string
   <td>Identifies the system or operating system
-   of the <a>endpoint node</a>.
- </tr>
+  of the <a>endpoint node</a>.
+  </tr>
 
- <tr>
+  <tr>
   <td><dfn>Platform version</dfn>
   <td>"<code>platformVersion</code>"
   <td>string
   <td>Identifies the system version of the <a>endpoint node</a>.
- </tr>
+  </tr>
 
- <tr>
+  <tr>
   <td><dfn>Accept SSL certificates</dfn>
   <td>"<code>acceptSslCerts</code>"
   <td>boolean
-  <td>Indicates that untrusted and self-signed SSL certificates
-   are trusted implicitly on <a data-lt=go>navigation</a>
-   for the duration of the session.
- </tr>
-</table>
-
-<p>Implementations are free to use the <a>capabilities</a> dictionary
- passed in the request to create a <a>new session</a>
- to provide additional feature requests
- from any of the <a>remote ends</a>, including the <a>endpoint node</a>.
- Likewise the <a>remote end</a> may supply additional entries
- on the <a>capabilities</a> object returned
- from creating a <a>new session</a>.
+  <td>Indicates whether untrusted and self-signed SSL certificates are
+  implicitly trusted on <a data-lt=go>navigation</a> for the duration of the
+  session.
```

Comment:
When discussing self-signed certificates at the last [F2F](https://www.w3.org/2016/07/14-webdriver-minutes.html#item02) there was talk from vendors that they/their infosec teams would not be too happy. We did discuss a mechanism for white listing domains for self signing, can't see it in the minutes, and @andreastt is looking to implement it in Marionette (we can hide how it works in geckodriver if need be). I think this is a good balance since @seleniumhq can overwrite it if it wants vendors still keep users safe